lib/md059.mjs

```diff
@@ -0,0 +1,60 @@
+// @ts-check
+
+import { addErrorContext } from "../helpers/helpers.cjs";
+import { filterByTypesCached } from "./cache.mjs";
+
+const defaultBannedText = [
+  "click here",
+  "here",
+  "learn more",
+  "link",
+  "more",
+  "read more"
+];
+
+/**
+ * Normalizes a string and removes extra whitespaces and punctuations.
+ *
+ * @param {string} text String to transform.
+ * @returns {string} Normalized string with no punctuations or extra whitespaces.
+ */
+function normalizeText(text) {
+  return text
+    .toLowerCase()
+    .replace(/\W+/g, " ")
+    .replace(/\s+/g, " ")
+    .trim();
+}
+
+/** @type {import("markdownlint").Rule} */
+export default {
+  "names": [ "MD059", "descriptive-link-text" ],
+  "description": "Link text should be descriptive",
+  "tags": [ "links", "accessibility" ],
+  "parser": "micromark",
+  "function": function MD059(params, onError) {
+    const bannedNames = new Set(params.config.link_texts || defaultBannedText);
+    const labels = filterByTypesCached([ "label" ])
+      .filter((label) => label.parent?.type === "link");
+
+    for (const label of labels) {
+      const labelTexts = label.children.filter((child) => child.type === "labelText");
+      for (const labelText of labelTexts) {
+        const { text } = label;
+        if (bannedNames.has(normalizeText(text))) {
+          const range = labelText.startLine === labelText.endLine ?
+            [ labelText.startColumn, text.length ] :
+            undefined;
+          addErrorContext(
+            onError,
+            labelText.startLine,
+            text,
```

Comment:
```
  -       errorContext: `[click␍␊
  +       errorContext: `[click␊
          here]`,
```

I can't figure out why [Windows behaves differently](https://github.com/DavidAnson/markdownlint/actions/runs/12840914345/job/35810130734?pr=1459) and returns a different `text` for the link test case with a line break. Curious if you might have any ideas @DavidAnson.

Reply:
This looks like a \r\n vs. \n line ending difference. If you're getting this error, it's reporting an error that spans multiple lines. I left a comment about preventing that by checking startLine and endLine of the span. (See my comment on the main chat before spending time on this.)